tensilelite/Tensile/KernelWriter.py

```diff
@@ -2852,6 +2847,11 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
 
     module.addComment1("Before NLL: Check VGPR.checkin for INT8 LW")
 
+    if kernel["PrefetchGlobalRead"] == 2 and (kernel["DirectToLdsA"] or kernel["DirectToLdsB"]):
```

Comment:
We should not need this here.
This will make prefetch less effective.
This should be before this code.
https://github.com/ROCm/hipBLASLt/blob/develop/tensilelite/Tensile/KernelWriter.py#L2183

I think you can refer Tensile code.
I can see wait is added at this location in Tensile case.

Reply:
We might have some bug fixes not ported from Tensile to tensilelite (especially DTL related).
Please make sure you checked logic in Tensile.

Reply:
Ah, thanks for the pointer! That is a better place in the NLL to place the wait instead of at the beginning.